### PR TITLE
Implement menu separators

### LINKED ISSUE #5999: Include an option to add a menu item separator. Sometimes menu items need to be logically grouped.
### Description

Include an option to add a menu item separator. Sometimes menu items need to be logically grouped according to their function.

### Public API Changes

Similar to `MenuFlyoutItem` and `MenuFlyoutSubItem`, this can be defined as `MenuFlyoutSeparatorItem` and it can be defined to denote where exactly this needs to be logically grouped.

```xaml
<MenuFlyoutItem Text="New" />
<MenuFlyoutItem Text="Open" />
<MenuFlyoutSeparatorItem />
<MenuFlyoutItem Text="Print" />
<MenuFlyoutSeparatorItem />
<MenuFlyoutItem Text="Exit" />
```

### Intended Use-Case

This adds a horizontal separator line to denote the logical grouping.

### PR DESCRIPTION
**Pull request by Eilon**
### Description of Change

This adds a new `MenuFlyoutSeparator` control that can be used like this:

```xaml
            <MenuFlyoutItem Text="Cool item 1"></MenuFlyoutItem>
            <MenuFlyoutSeparator />
            <MenuFlyoutItem Text="Cool item 2"></MenuFlyoutItem>
```

It has no events or properties because it doesn't _do_ anything - it just causes a horizontal line to be drawn between menu items.

Windows looks like this:
image

MacCatalyst looks like this:

image

Also part of this PR: I found some 100% unused separator-related types in MAUI. Looks like they were there for approximately one million years and became relics. So, I'm deleting those types so that they don't confuse future folks. (I found them while I was searching for `separator` and for a brief moment thought it was already implemented!)

### Issues Fixed

Fixes #5999

### CODE REVIEW

**Comment by Eilon** [comment]
I believe this is ready for review!

File: src/Controls/src/Core/SeparatorMenuItem.cs [hunk 1]
```diff
@@ -1,6 +0,0 @@
-namespace Microsoft.Maui.Controls
-{
-	internal class SeparatorMenuItem : BaseMenuItem
```
Comment (Eilon) [inline src/Controls/src/Core/SeparatorMenuItem.cs:3]: This type was internal and not used anywhere. Probably a legacy internal type from Xamarin.Forms that never got deleted.

File: src/Controls/src/Core/Handlers/Shell/Windows/ShellFlyoutSeparator.cs [hunk 2]
```diff
@@ -1,12 +0,0 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace Microsoft.Maui.Controls.Platform
-{
-	internal struct ShellFlyoutSeparator
```
Comment (Eilon) [inline src/Controls/src/Core/Handlers/Shell/Windows/ShellFlyoutSeparator.cs:9]: This type was internal and not used anywhere. Probably a legacy internal type from Xamarin.Forms that never got deleted.

**Comment by Eilon** [comment]
Also, I contemplated adding some kind of test, but I don't know what meaningful test I could add. This control does _nothing_ on its own; it only creates a native element when requested to do so. It has no properties, no methods, etc.

**Comment by Eilon** [comment]
@PureWeen / @mattleibow - any idea who would be able to review this? It's less important than servicing stuff, but would be nice for .NET 7.

File: src/Core/src/Handlers/MenuFlyoutSeparator/MenuFlyoutSeparatorHandler.iOS.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using UIKit;
+
+namespace Microsoft.Maui.Handlers
+{
+	[System.Runtime.Versioning.SupportedOSPlatform("ios13.0")]
```
[the remark this answers is not in the document]
Comment (Eilon) [reply src/Core/src/Handlers/MenuFlyoutSeparator/MenuFlyoutSeparatorHandler.iOS.cs:6]: I think they already all have it (I think I just copied what they were doing):

* https://github.com/dotnet/maui/blob/dc8cce85ca06eac200f0c9e7db036d0bd6fa2231/src/Core/src/Handlers/MenuBar/MenuBarHandler.iOS.cs#L11
* https://github.com/dotnet/maui/blob/dc8cce85ca06eac200f0c9e7db036d0bd6fa2231/src/Core/src/Handlers/MenuBarItem/MenuBarItemHandler.iOS.cs#L10
* https://github.com/dotnet/maui/blob/dc8cce85ca06eac200f0c9e7db036d0bd6fa2231/src/Core/src/Handlers/MenuFlyoutItem/MenuFlyoutItemHandler.iOS.cs#L10
* https://github.com/dotnet/maui/blob/dc8cce85ca06eac200f0c9e7db036d0bd6fa2231/src/Core/src/Handlers/MenuFlyoutSubItem/MenuFlyoutSubItemHandler.iOS.cs#L11
* https://github.com/dotnet/maui/blob/dc8cce85ca06eac200f0c9e7db036d0bd6fa2231/src/Core/src/Handlers/MenuFlyoutSeparator/MenuFlyoutSeparatorHandler.iOS.cs#L6 (this PR)

Or do you mean another thing?

[the remark this answers is not in the document]
Comment (PureWeen) [reply src/Core/src/Handlers/MenuFlyoutSeparator/MenuFlyoutSeparatorHandler.iOS.cs:6]: Oh snap! I wrote all these originally and neglected to check if someone had already added these. 

My bad
Looks good!

**Review by PureWeen — APPROVED** [review]

**Pull request merged**